src/main/java/edu/harvard/iq/dataverse/dataverse/validation/BannerErrorHandler.java

```diff
@@ -0,0 +1,88 @@
+package edu.harvard.iq.dataverse.dataverse.validation;
+
+import edu.harvard.iq.dataverse.dataverse.banners.BannerLimits;
+import edu.harvard.iq.dataverse.dataverse.banners.DataverseBanner;
+import edu.harvard.iq.dataverse.dataverse.banners.DataverseLocalizedBanner;
+import edu.harvard.iq.dataverse.util.BundleUtil;
+import edu.harvard.iq.dataverse.util.DataverseClock;
+import edu.harvard.iq.dataverse.util.DateUtil;
+
+import javax.ejb.Stateless;
+import javax.faces.application.FacesMessage;
+import javax.faces.context.FacesContext;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Class which handles errors when adding new banner, errors are displayed thanks to <p:message/>
+ */
+@Stateless
+public class BannerErrorHandler {
+
+    public List<FacesMessage> handleBannerAddingErrors(DataverseBanner banner,
+                                                       DataverseLocalizedBanner dlb,
+                                                       FacesContext faceContext) {
+        ArrayList<DataverseLocalizedBanner> dataverseLocalizedBanners =
+                new ArrayList<>(banner.getDataverseLocalizedBanner());
+
+        if (dlb.getImage().length < 1) {
+            throwImageWasMissing(faceContext, dataverseLocalizedBanners.indexOf(dlb));
+
+        } else if (ImageValidator.isImageResolutionTooBig(dlb.getImage(),
+                BannerLimits.MAX_WIDTH.getValue(), BannerLimits.MAX_HEIGHT.getValue())) {
+            throwResolutionTooBigError(faceContext, dataverseLocalizedBanners.indexOf(dlb));
+        }
+
+        if (dlb.getImage().length > BannerLimits.MAX_SIZE_IN_BYTES.getValue()) {
+            throwSizeTooBigError(faceContext, dataverseLocalizedBanners.indexOf(dlb));
```

Comment:
missleading method names `throw*` they do not throw anything

Reply:
Yeah, that's true in the java context it will not throw anything how about `addErrorMessage*`